File: bekend/src/main/java/com/application/bekend/service/AdditionalServicesService.java
package com.application.bekend.service;

import com.application.bekend.model.AdditionalServices;
import com.application.bekend.repository.AdditionalServicesRepository;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Set;

@Service
public class AdditionalServicesService {

    private final AdditionalServicesRepository additionalServicesRepository;

    public AdditionalServicesService(AdditionalServicesRepository additionalServicesRepository) {
        this.additionalServicesRepository = additionalServicesRepository;
    }

    public AdditionalServices save(AdditionalServices additionalServices) {
        return this.additionalServicesRepository.save(additionalServices);
    }

    public List<AdditionalServices> getAllByHouseId(Long id) { return additionalServicesRepository.getAllByHouseId(id); };

    public List<AdditionalServices> getAllByBoatId(Long id) { return additionalServicesRepository.getAllByBoatId(id); };

    public Set<AdditionalServices> getAllByHouseReservationId(Long id) { return additionalServicesRepository.getAllByHouseReservationId(id); }

    public AdditionalServices getAdditionalServicesById(Long id){return additionalServicesRepository.getAdditionalServicesById(id);}
    
    public List<AdditionalServices> getAllByFishingAdventureId(Long id) { return additionalServicesRepository.getAllByFishingAdventureId(id); };
}

File: frontend/src/app/service/additional-services.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs";
import {Room} from "../model/room";
import {AdditionalService} from "../model/additional-service";
import {HouseReservation} from "../model/house-reservation";
import {environment} from "../../environments/environment";

@Injectable({
  providedIn: 'root'
})
export class AdditionalServicesService {

  constructor(private _http:HttpClient) { }

  private readonly userPath = environment.backend_api + 'api/additionalServices';

  public getAllByHouseId(id: number): Observable<AdditionalService[]>{
    return this._http.get<AdditionalService[]>(`${this.userPath}/getAllByHouseId/`+id)
  }

  public getAllByBoatId(id: number): Observable<AdditionalService[]>{
    return this._http.get<AdditionalService[]>(`${this.userPath}/getAllByBoatId/`+id)
  }

  public getAllByHouseReservationId(id: number): Observable<AdditionalService[]>{
    return this._http.get<AdditionalService[]>(`${this.userPath}/getAllByHouseReservationId/`+id)
  }

  public getAllByFishingAdventureId(id: number): Observable<AdditionalService[]>{
    return this._http.get<AdditionalService[]>(`${this.userPath}/getAllByFishingAdventureId/`+id)
  }

  public getAllByBoatReservationId(id: number): Observable<AdditionalService[]>{
    return this._http.get<AdditionalService[]>(`${this.userPath}/getAllByBoatReservationId/`+id)
  }

  public delete(id: number): Observable<boolean> {
    return this._http.delete<boolean>(`${this.userPath}/delete/`+id)
  }

  public save(additionalService: AdditionalService): Observable<AdditionalService> {
    return this._http.post<AdditionalService>(`${this.userPath}/add`, additionalService)
  }
}
